libcontainerd/client_local_windows.go

```diff
@@ -299,6 +300,78 @@ func (c *client) createWindows(id string, spec *specs.Spec, runtimeOptions inter
 		}
 	}
 	configuration.MappedDirectories = mds
+
+	// Proxy Support for RS4
+	if system.GetOSVersion().Build > 17120 {
+		proxy := ""
+		exceptions := ""
+
+		for _, envVar := range spec.Process.Env {
+			if strings.HasPrefix(envVar, "HTTP_PROXY") ||
+				strings.HasPrefix(envVar, "HTTPS_PROXY") ||
+				strings.HasPrefix(envVar, "FTP_PROXY") {
+				proxy = strings.Split(envVar, "=")[1]
+			} else if strings.HasPrefix(envVar, "NO_PROXY") {
+				exceptions = strings.Replace(strings.Split(envVar, "=")[1], ",", ";", -1)
+			}
+		}
+
+		if proxy != "" {
+			defaultConnectionString := []byte{
+				0x46, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,
```

Comment:
What are these? Is there an MSDN link or something? (More for interest, not questioning correctness). If there's documentation, might be worth adding a link to it.

Reply:
At the time I added these I remember having used MSDN for figuring this out. I could find these documents https://support.microsoft.com/en-us/help/900935/how-the-windows-update-client-determines-which-proxy-server-to-use-to and https://securelink.be/blog/windows-proxy-settings-explained/ but didn't find any documents for the exact registry settings documented